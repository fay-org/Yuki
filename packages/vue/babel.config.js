module.exports = {
  presets: ['@fect-ui/cli/preset'],
}
