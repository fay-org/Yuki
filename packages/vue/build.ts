import { BuildTaskConfig, TASK_NAME, runTask, commonOutput } from 'internal'
import { createBundle } from 'no-bump'

const { build } = createBundle({})

const commonInput = 'src/index.ts'

/**
 * I decide split the build task into four.
 * First is only bundle script module.
 * Second part only parser style moudle.
 * Thrid part will generator all analyzed and generator style deps graph.
 * Fourth part will generator the entry file as umd.
 *
 * The first task will have two sub task. with esm task coomonjs task
 */
