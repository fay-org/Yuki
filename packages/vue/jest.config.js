module.exports = {
  testPathIgnorePatterns: ['/node_modules/', '/dist/', '/lib/', '/es/'],
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx,vue}',
    '!src/utils/**',
    '!src/index.ts',
    '!src/**/type.ts',
    '!src/**/style.ts',
    '!src/**/interface.ts',
    '!src/back-top/**'
  ],
  collectCoverage: true,
  preset: '@fect-ui/cli/lib/config',
  testTimeout: 20000
}
