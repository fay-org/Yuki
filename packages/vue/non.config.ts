import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  lib: {
    input: path.join(__dirname, 'components'),
    format: ['cjs', 'es', 'umd'],
    name: 'fect'
  }
})
