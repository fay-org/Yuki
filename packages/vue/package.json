{
  "name": "@fect-ui/vue",
  "version": "1.4.0-rc.1",
  "description": "Minimalist UI components built on Vue",
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "types/index.d.ts",
  "jsdelivr": "lib/fect.min.umd.js",
  "scripts": {
    "collect": "non collect",
    "build": "npm run collect && non compile && non volar && node ./typing-template.js",
    "test": "non test"
  },
  "keywords": [
    "vue component",
    "fect",
    "vue ui"
  ],
  "homepage": "https://github.com/fect-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fect-org/fect"
  },
  "files": [
    "lib/",
    "es/",
    "types/",
    "vetur/",
    "README.md",
    "components.d.ts"
  ],
  "author": "XeryYue",
  "license": "MIT",
  "devDependencies": {
    "@fect-ui/cli": "^1.0.0"
  },
  "sideEffects": [
    "es/**/style/*",
    "lib/**/style/*",
    "*.css"
  ],
  "dependencies": {
    "@fect-ui/themes": "^2.2.0",
    "@fect-ui/vue-hooks": "^1.2.0",
    "proy": "^0.0.3",
    "@popperjs/core": "^2.11.2"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  }
}
