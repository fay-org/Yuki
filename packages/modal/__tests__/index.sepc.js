import { mount } from '@vue/test-utils'
import { ref } from 'vue'
import Modal from '..'

describe('Modal', () => {
  it('should be render as element', () => {
    const wrapper = mount(Modal, {
      props: {
        visible: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('props should be supported', () => {
    const wrapper = mount(Modal, {
      props: {
        title: 'Test Modal',
        cancel: 'Cancel',
        done: 'Done',
        visible: true,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.modal_title').text()).toBe('Test Modal')
    expect(wrapper.findAll('.modal_action__btn')[0].text()).toBe('Cancel')
    expect(wrapper.findAll('.modal_action__btn')[1].text()).toBe('Done')
  })
  it('should support slot to define custom title and action ', () => {
    const wrapper = mount({
      setup() {
        return {
          customSlots: {
            title: () => 'Custom Title',
            action: () => <h3>Custom Action</h3>,
          },
        }
      },
      render() {
        return <Modal v-slots={this.customSlots} visible={true}></Modal>
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-modal_title__container').text()).toBe(
      'Custom Title',
    )
    expect(wrapper.find('h3').text()).toBe('Custom Action')
  })
  it('should be close modal through props visible value is false', async () => {
    const wrapper = mount({
      setup() {
        const show = ref(false)
        const handlerClick = () => (show.value = true)
        return { show, handlerClick }
      },
      render() {
        return (
          <div>
            <button id="btn" onClick={this.handlerClick}>
              Button
            </button>
            <Modal title="Test" v-model={[this.show, 'visible']}></Modal>
          </div>
        )
      },
    })
    const btn = wrapper.find('#btn')
    expect(wrapper.vm.show).toBe(false)
    await btn.trigger('click')
    expect(wrapper.vm.show).toBe(true)
    await expect(wrapper.findAll('.modal_action__btn')[0].trigger('click'))
    expect(wrapper.vm.show).toBe(false)
  })
})
