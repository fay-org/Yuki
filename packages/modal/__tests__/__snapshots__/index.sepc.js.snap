// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal props should be supported 1`] = `
<transition-stub>
  <div class="fect-dialog_root">
    <div class="fect-dialog-backdrop"></div>
    <div class="fect-dialog-backdrop responsive"></div>
    <transition-stub>
      <div role="dialog" class="fect-modal_wrapper" style="width: 400px;">
        <div class="fect-modal_title__container">
          <h2 class="modal_title">Test Modal</h2>
        </div>
        <div class="fect-modal_content">
          <!---->
        </div>
        <footer class="fect-modal_action__container"><button class="fect-button fect-button--default fect-button--medium modal_action__btn" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);">
            <!---->
            <!---->
            <!---->Cancel
          </button><button class="fect-button fect-button--default fect-button--medium modal_action__btn" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);">
            <!---->
            <!---->
            <!---->Done
          </button></footer>
      </div>
    </transition-stub>
  </div>
</transition-stub>
`;

exports[`Modal should be render as element 1`] = `
<transition-stub>
  <div class="fect-dialog_root">
    <div class="fect-dialog-backdrop"></div>
    <div class="fect-dialog-backdrop responsive"></div>
    <transition-stub>
      <div role="dialog" class="fect-modal_wrapper" style="width: 400px;">
        <div class="fect-modal_title__container">
          <h2 class="modal_title"></h2>
        </div>
        <div class="fect-modal_content">
          <!---->
        </div>
        <footer class="fect-modal_action__container"><button class="fect-button fect-button--default fect-button--medium modal_action__btn" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);">
            <!---->
            <!---->
            <!---->cancel
          </button><button class="fect-button fect-button--default fect-button--medium modal_action__btn" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);">
            <!---->
            <!---->
            <!---->done
          </button></footer>
      </div>
    </transition-stub>
  </div>
</transition-stub>
`;

exports[`Modal should support slot to define custom title and action  1`] = `
<transition-stub>
  <div class="fect-dialog_root">
    <div class="fect-dialog-backdrop"></div>
    <div class="fect-dialog-backdrop responsive"></div>
    <transition-stub>
      <div role="dialog" class="fect-modal_wrapper" style="width: 400px;">
        <div class="fect-modal_title__container">Custom Title</div>
        <div class="fect-modal_content">
          <!---->
        </div>
        <h3>Custom Action</h3>
      </div>
    </transition-stub>
  </div>
</transition-stub>
`;
